### PR TITLE
[i1050] Add missing test coverage for medication_refill_helper.rb

### PR DESCRIPTION
**Pull request by HolyWalley**
# Description 

Add tests for the medication_refill_helper.rb

## Corresponding Issue

https://github.com/ifmeorg/ifme/issues/1050

# Test Coverage

✅

### CODE REVIEW

File: spec/helpers/calendar_helper_spec.rb [hunk 1]
```diff
@@ -9,7 +9,7 @@
     it 'when add_to_google_cal is true without a refill date' do
       medication.add_to_google_cal = true
       medication.refill = nil
-      expect(new_cal_refill_reminder_needed?(medication)).to eq(false)
+      expect(new_cal_refill_reminder_needed?(medication)).to eq(nil)
```
Comment (julianguyen) [inline spec/helpers/calendar_helper_spec.rb]: Why was this changed to nil?

File: spec/helpers/medication_refill_helper_spec.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    let(:user) { FactoryBot.create(:user1) }
+    let(:medication) { FactoryBot.create(:medication, user_id: user.id) }
+    let(:exception_text) { 'RESCUE INVOKED' }
+
+    before do
+      allow_any_instance_of(helper.class).to receive(:return_to_sign_in).and_return(exception_text)
+      sign_in user
+    end
+
+    context 'when current_user.google_oauth2_enabled? && new_cal_refill_reminder_needed?(medication)' do
```
Comment (julianguyen) [inline spec/helpers/medication_refill_helper_spec.rb]: I would write this out as a sentence instead of referring to methods in the context

File: spec/helpers/medication_refill_helper_spec.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+      before do
+        allow_any_instance_of(User).to receive(:google_oauth2_enabled?).and_return(true)
+        allow_any_instance_of(helper.class).to receive(:new_cal_refill_reminder_needed?).and_return(true)
+      end
+
+      it { expect(helper.save_refill_to_google_calendar(medication)).to eq(exception_text) }
+    end
+
+    context 'until current_user.google_oauth2_enabled? && new_cal_refill_reminder_needed?(medication)' do
+      it { expect(helper.save_refill_to_google_calendar(medication)).to eq(true) }
```
Comment (julianguyen) [inline spec/helpers/medication_refill_helper_spec.rb]: Similar comment as above

File: spec/helpers/calendar_helper_spec.rb (hunk 1 above)
Comment (HolyWalley) [reply spec/helpers/calendar_helper_spec.rb]: so, in ruby nil is a negative value in bool expressions. for example

if nil
  true
else
  false
end

#=> false

but nil != false and this is why test is falling down with false instead of nil.

File: spec/helpers/medication_refill_helper_spec.rb (hunk 2 above)
Comment (HolyWalley) [reply spec/helpers/medication_refill_helper_spec.rb]: I agree. I'll fix it in a few hours

File: spec/helpers/calendar_helper_spec.rb (hunk 1 above)
Comment (HolyWalley) [reply spec/helpers/calendar_helper_spec.rb]: we can also change `#new_cal_refill_reminder_needed` to this:

```
!!(medication.add_to_google_cal && medication.refill &&
  (medication.add_to_google_cal_changed? || medication.refill_changed?))
```

it will have a similar result, but I'm not sure that this is good practice.

Comment (julianguyen) [reply spec/helpers/calendar_helper_spec.rb]: Hmm, ideally that method should return a boolean. Returning either true or nil isn't very consistent to me. 

Double nots are messy too, as you pointed out, what about `nil?`

File: spec/helpers/medication_refill_helper_spec.rb [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    let(:user) { FactoryBot.create(:user1) }
+    let(:medication) { FactoryBot.create(:medication, user_id: user.id) }
+    let(:exception_text) { 'RESCUE INVOKED' }
+
+    before do
+      allow_any_instance_of(helper.class).to receive(:return_to_sign_in).and_return(exception_text)
+      sign_in user
+    end
+
+    context 'when the user has the google oauth2 enabled and he need a new refill reminder needed' do
```
Comment (julianguyen) [inline spec/helpers/medication_refill_helper_spec.rb]: Let's remove the "he" pronoun to be neutral. You can replace it with "they" instead.

File: spec/helpers/medication_refill_helper_spec.rb (hunk 3 above)
Comment (julianguyen) [reply spec/helpers/medication_refill_helper_spec.rb]: Similar pronoun comment above

**Comment by HolyWalley** [comment]
@julianguyen done :)
